File: Day_0924/Day_0924/exam_31.c
#include<stdio.h>
struct person {
	int bun;
	char name[20];
	int score;
};
main() {
	struct person a[] = { {1,"aaa",90},{2,"bbb",88},{3,"ccc",95} };
	int s = 0;
	for (int i = 0; i < 3; i++) {
		s += a[i].score;
	}
	printf("%d\n",s); // 출력값 : 273
	// 구조체 포인터
	struct person* p;
	p = a;
	printf("bun=%d\n", a[0].bun);
	printf("bun=%d\n", p->bun);
	p++;
	printf("bun=%d\n", a[1].bun);
	printf("bun=%d\n", p->bun);

	printf("bun=%d\n", a[2].bun);
	printf("bun=%d\n", (p+1)->bun);
}

File: Day_1001/exam31.py
# 상속

class AA:
    def __init__(self, bun, name):
        self.bun = bun
        self.name = name

    def aPrint(self):
        print('bun = ', self.bun)
        print('name = ', self.name)

    def exam(self):
        print('AA Class')
        for x in range(6):
            print(x, end = '\t')
        print()

class Sub_AA(AA):
    def __init__(self, bun, name):
        super().__init__ (bun,name)
    
    def test(self):
        print('Sub_AA Class')
        for x in range(5):
            print(x, end = '\t')
        print()
    

aa = AA(1,'aaa')
aa.aPrint()
aa.exam()
print('-----------------------------')

sub_bb = Sub_AA(5,'bbb')
sub_bb.aPrint()
sub_bb.exam()
sub_bb.test()
